nemo/collections/nlp/data/datasets/text_classification/text_classification_data_descriptor.py

```diff
@@ -0,0 +1,101 @@
+from nemo import logging
+from nemo.collections.nlp.data.datasets.datasets_utils import (
+    calc_class_weights,
+    get_intent_labels,
+    get_label_stats,
+    if_exist,
+    process_imdb,
+    process_jarvis_datasets,
+    process_nlu,
+    process_sst_2,
+    process_thucnews,
+)
+
+__all__ = ['TextClassificationDataDesc']
+
+
+class TextClassificationDataDesc:
+    def __init__(self, dataset_name, data_dir, do_lower_case, eval_file_prefix=None):
+
+        if dataset_name == 'sst-2':
+            self.data_dir = process_sst_2(data_dir)
+            self.num_labels = 2
+            self.eval_file = self.data_dir + '/dev.tsv'
+
+        elif dataset_name == 'imdb':
+            self.num_labels = 2
+            self.data_dir = process_imdb(data_dir, do_lower_case)
+            self.eval_file = self.data_dir + '/test.tsv'
+        elif dataset_name == 'thucnews':
+            self.num_labels = 14
+            self.data_dir = process_thucnews(data_dir)
+            self.eval_file = self.data_dir + '/test.tsv'
+        elif dataset_name.startswith('nlu-'):
+            if dataset_name.endswith('chat'):
+                self.data_dir = f'{data_dir}/ChatbotCorpus.json'
+                self.num_labels = 2
+            elif dataset_name.endswith('ubuntu'):
+                self.data_dir = f'{data_dir}/AskUbuntuCorpus.json'
+                self.num_labels = 5
+            elif dataset_name.endswith('web'):
+                data_dir = f'{data_dir}/WebApplicationsCorpus.json'
+                self.num_labels = 8
+            self.data_dir = process_nlu(data_dir, do_lower_case, dataset_name=dataset_name)
+            self.eval_file = self.data_dir + '/test.tsv'
+        elif dataset_name.startswith('jarvis'):
+            self.data_dir = process_jarvis_datasets(
+                data_dir, do_lower_case, dataset_name, modes=['train', 'test', 'eval'], ignore_prev_intent=False
+            )
+
+            intents = get_intent_labels(f'{self.data_dir}/dict.intents.csv')
+            self.num_labels = len(intents)
+        else:
```

Comment:
What happens when user wants to try their own dataset? Do we have a format that they can use?

Reply:
added default_format